src/main/kotlin/org/jetbrains/research/testspark/tools/llm/Llm.kt

```diff
@@ -32,27 +34,38 @@ class Llm(override val name: String = "LLM") : Tool {
      * @param testSamplesCode The test samples code.
      * @return An instance of LLMProcessManager.
      */
-    private fun getLLMProcessManager(project: Project, psiFile: PsiFile, caretOffset: Int, codeType: FragmentToTestData, testSamplesCode: String): LLMProcessManager {
+    private fun getLLMProcessManager(
+        project: Project,
+        psiFile: PsiFile,
+        caretOffset: Int,
+        codeType: FragmentToTestData,
+        testSamplesCode: String,
+    ): LLMProcessManager {
         val classesToTest = mutableListOf<PsiClass>()
         // check if cut has any none java super class
         val maxPolymorphismDepth = SettingsArguments.maxPolyDepth(0)
 
         val cutPsiClass: PsiClass = getSurroundingClass(psiFile, caretOffset)!!
 
-        var currentPsiClass = cutPsiClass
-        for (index in 0 until maxPolymorphismDepth) {
-            if (!classesToTest.contains(currentPsiClass)) {
-                classesToTest.add(currentPsiClass)
-            }
+        ApplicationManager.getApplication().runReadAction(
```

Comment:
This code snippet is getting bigger. Maybe it can be a function for better readability